mephisto/client/review/review_server.py

```diff
@@ -147,36 +159,89 @@ def data():
             {"finished": finished, "data": current_data if not finished else None}
         )
 
-    @app.route("/all_data")
-    def all_task_data():
-        if not all_data and database_task_name is None:
+    @app.route("/data/<id>", methods=["GET", "POST"])
+    def task_data_by_id(id):
+        global finished, current_data, ready_for_next, counter, all_data_list
+        id = int(id) if type(id) is int or (type(id) is str and id.isdigit()) else None
+        if request.method == "GET":
+            if all_data:
+                list_len = len(all_data_list)
+                if id is None or id < 0 or id >= list_len:
+                    return jsonify(
+                        {"error": f"Data with ID: {id} does not exist", "mode": MODE}
+                    )
+                return jsonify({"data": all_data_list[id], "mode": MODE})
+            else:
+                if id is None or id != counter - 1:
+                    return jsonify(
+                        {
+                            "error": f"Please review the data point with ID: {counter - 1}",
+                            "mode": MODE,
+                        }
+                    )
+                if finished:
+                    func = request.environ.get("werkzeug.server.shutdown")
+                    if func is None:
+                        raise RuntimeError("Not running with the Werkzeug Server")
+                    func()
+                return jsonify(
+                    {
+                        "finished": finished,
+                        "data": current_data if not finished else None,
+                        "mode": MODE,
+                    }
+                )
+        else:
+            review = request.get_json(force=True)
+            if output == "":
+                sys.stdout.write("{}\n".format(review))
```

Comment:
Is there a specific reason why you're directly writing with sys rather than print or some other kind of logging.

Reply:
I just used the method originally implemented in this app to write out reviews. I can change the method for both this instance and other instances if there is a more efficient way.

Reply:
@pringshia was there a reason you were using `sys.stdout` particularly? Otherwise we can likely just use `print`.

Reply:
no particular reason @JackUrb aside from the fact that since originally this was demoed as a unix-esque CLI use-case of piping back out to shell, i was trying to be symmetrical to reading in from `sys.stdin`. After doing some research, I see that there's no other reason between either usage, except that `print` adds a newline where as `sys.stdout.write` doesn't. Since we're adding on a new line at the end anyway, it seems fine to use print to me.

Reply:
Sounds good, I'll make the changes